packages/react/menu/src/Menu.tsx

```diff
@@ -652,10 +653,24 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
       menuItem.addEventListener(ITEM_SELECT, (event) => onSelect?.(event), { once: true });
       menuItem.dispatchEvent(itemSelectEvent);
       if (itemSelectEvent.defaultPrevented) return;
-      context.onRootClose();
+      /**
+       * Browsers wait for our handler logic to execute before executing their own implementation
+       * in case we have prevented the default behaviour. Therefore, the root will unmount before
+       * the browser's `pointerup` logic has executed causing `pointerdown` to be stuck in some
+       * browsers. In FF this results in text selection occuring on the page when moving the mouse
+       * after close.
+       *
+       * We move the close logic till after the browser receives the pointerup to avoid this.
+       */
```

Comment:
Wondering whether we need to adjust this comment given that this fixes other issues https://github.com/radix-ui/primitives/pull/812#issuecomment-891889618

Reply:
Interesting... It feels a bit odd to comment here that we delay to allow composition with `onClick` because that means we're baking logic into components for external dependencies when components shouldn't really have knowledge of, or handling for, external things.

I can't actually get the `DropdownMenu.Item` + `Dialog.Trigger` to work locally with this change though (I made some tweaks to the code so that the timeout only applies where needed).

Reply:
Ah, I can get it to work... ignore me.

Reply:
It seems that the previous way I wrote this still didn't fix the composition issue for keyboard users but now I'm a bit hesitant to push this change given that it masks that other issue a bit.